plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,441 @@
+    """Parses the general.keystore file.
+
+    Args:
+      file_entry (dfvfs.FileEntry): the OneDrive Log file entry.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+
+    Returns:
+      bytes: the key, or None if the general.keystore file could not be found or
+          parsed.
+    """
+    general_keystore_entry = self._OpenFileInSameDirectory(
+        file_entry, self._GENERAL_KEYSTORE_FILE, parser_mediator)
+
+    if not general_keystore_entry:
+      return None
+
+    general_keystore_file_object = general_keystore_entry.GetFileObject()
+    general_keystore_data = general_keystore_file_object.read()
+    json_data = json.loads(general_keystore_data)
+
+    if not json_data:
+      parser_mediator.ProduceExtractionWarning(
+          'Unable to parse OneDrive keystore file {0:s}'.format(
+              general_keystore_entry.name))
+      return None
+
+    key = json_data[0].get('Key', None)
+    decoded_key = base64.b64decode(key)
+
+    if len(decoded_key) != 32:
+      return None
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    delimiters = r'(' + r'|'.join(self._DELIMITERS) + r')'
+
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = re.split(delimiters, extracted_string)
+      output_string = []
+      for token in tokens:
+        if token not in self._DELIMITERS:
+          token = obfuscated_string_map.get(token, token)
+        output_string.append(token)
+      output_strings.append(''.join(output_string))
+
+    return output_strings
+
+  def _DecryptStrings(self, extracted_strings, aes_key):
+    """Decrypt strings extracted from OneDrive logs.
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and a null initialization vector.
+    * base64 encoded where the resulting '/' and '+'' characters are replaced
+      with '_' and '-'' respectively.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the list of decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+
+    base64_alphabet = (
+        'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-')
+
+    delimiters = r'(' + r'|'.join(self._DELIMITERS) + r')'
```

Comment:
Since you're using this multiple times why not make this a class constant?

Reply:
Done.